astrokat/observe_main.py

```diff
@@ -258,217 +263,250 @@ def __enter__(self):
         return self
 
     def __exit__(self, type, value, traceback):
-        user_logger.info('Returning telescope to startup state')
+        """Return telescope to its startup state."""
+        user_logger.info("Returning telescope to startup state")
         # Ensure known exit state before quitting
         # TODO: Return correlator settings to entry values
         # switch noise-source pattern off (ensure this after each observation)
         noisediode.off(self.array)
         self.array.disconnect()
 
     def subarray_setup(self, instrument):
-        # current sensor list included in instrument
-        # sub_ [
-                # pool_resources,  # ptuse or specific antennas
-                # product,        # correlator product
-                # dump_rate,      # dumprate
-                # band,           # band
-                # ]
-        user_logger.trace(self.opts.obs_plan_params['instrument'])
-        if self.opts.obs_plan_params['instrument'] is None:
+        """Set up the array for observing.
+
+        Include current sensor list in instrument.
+
+        Parameters
+        ----------
+        instrument: object
```

Comment:
`object` is very vague.  From the code looks like it is a `dict`, or at least `dict-like`.